tensorflow/core/grappler/optimizers/generic_layout_optimizer.cc

```diff
@@ -109,10 +109,17 @@ inline std::pair<string, string> GetSrcAndDstDataFormats(
     const TransposeContext& context, int num_gpus, int num_voltas) {
   string src_format = kNHWC;
   string dst_format = kNCHW;
-  if (((static_cast<float>(num_voltas) / static_cast<float>(num_gpus)) >=
-       kVoltaGPURatioThreshold) &&
-      NumConvOnDeviceWithDataTypeOverThreshold(context, kGPU, DT_HALF)) {
-    std::swap(src_format, dst_format);
+  if (!context.enforced_layout.empty()) {
+    if (context.enforced_layout == "NHWC") {
+      std::swap(src_format, dst_format);
+    }
+    // else, enforced_layout == "NCHW", as it was previously checked
```

Comment:
Let's move this comment line into the else-branch, and revise it to:

// Otherwise, enforced_layout is NCHW enforced by previous check.